.github/workflows/lint-code.yaml

```diff
@@ -0,0 +1,37 @@
+name: Lint Code plus JS Unit Tests
+
+# This workflow checks to make sure that all dependency binaries are valid, to prevent a sneaky commit doing something malicious
+on:
+  pull_request:
+    branches: ['**', '**']
+  push:
+    branches: [master]
+
+jobs:
+  lint-code:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Setup Environment
+      uses: ./.github/workflows/composite/setup-docker-ci
+      with:
+        merge_master: ${{ matrix.merge == 'merged' }}
+        skip_docker: 'false'
+        event_name: ${{ github.event_name }}
+
+    - name: dependencies
+      env:
+        IS_LUNASEC_CI: true
+      run: yarn install --immutable --immutable-cache --inline-builds
```

Comment:
Why are we doing this bare-metal install?  This should be happening in the docker container, called in the above setup step right?  If this is the security "check" we talked about before, maybe we should just put that in its own step since it has nothing to do with any of the rest of this stuff and doesn't depend on docker.